### PR TITLE
Update generics.md

### PR DESCRIPTION
**Pull request by Vikaton**
got it from the reddit post, decided to make it public, credit goes to Exilor

### CODE REVIEW

File: _gitbook/syntax_and_semantics/generics.md [hunk 1]
```diff
@@ -1 +1,78 @@
...
+```
+class Hash(K, V)
+end
+```
+In this case K stands for the type of the keys, and V for the type of the values.
+```
+hash = Hash(Symbol, String) # or "{} of Symbol => String"
+hash[:name] = "Exilor"
+```
+Range too used to have two and will have two again when 0.6.2 is released:
```
Comment (jhass) [inline _gitbook/syntax_and_semantics/generics.md]: I think at least this section needs some rephrasing. The text is generally rather targeted at Rubyists, I'm not sure if that's well suited for the official docs tbh.

Comment (asterite) [inline _gitbook/syntax_and_semantics/generics.md]: Yes, I wouldn't mention Ruby at all in the docs. Right now it's mentioned only once, saying that it's syntax is similar to Ruby. The docs should explain the language without assuming that you know Ruby or other languages (although that's kind of impossible).

I also wouldn't use existing classes from the standard library to explain generics, but a separate class to explain the concepts little by little.

Comment (asterite) [inline _gitbook/syntax_and_semantics/generics.md]: I'll try to reuse what you have written and what I know about generics to write this page. There are many things to explain, for example type argument inference from new/initialize, and why there's a need for generics.

Comment (jhass) [inline _gitbook/syntax_and_semantics/generics.md]: @asterite to make it clear, it's a copy paste from http://www.reddit.com/r/crystal_programming/comments/314xqy/about_generic_classes_and_modules/ by @Exilor

**Comment by asterite** [comment]
I already documented that section, so I'll close this.

**Pull request closed without merging**